saleor/core/utils/__init__.py

```diff
@@ -51,10 +52,37 @@ def build_absolute_uri(location):
     return iri_to_uri(location)
 
 
+def is_valid_ipv4(ip: str) -> bool:
+    """Check whether the passed IP is a valid V4 IP address."""
+    try:
+        socket.inet_pton(socket.AF_INET, ip)
+    except socket.error:
+        return False
+    return True
+
+
+def is_valid_ipv6(ip: str) -> bool:
+    """Check whether the passed IP is a valid V6 IP address."""
+    try:
+        socket.inet_pton(socket.AF_INET6, ip)
+    except socket.error:
+        return False
+    return True
+
+
 def get_client_ip(request):
+    """Retrieve the IP address from the request data.
+
+    Tries to get the IP address from X-Forwarded-For, if the user is hiding behind
+    a transparent proxy. Otherwise, retrieve the requester IP.
+
+    If the forwarded IP is invalid, it fallback to the requester IP.
```

Comment:
Most of the time this will return the IP address of our own reverse proxy (eg. CloudFront).

Reply:
Are you asking about changing the comment or?

Reply:
I'm talking about this fallback not being something we want. We should fall back to the first valid IP in the "forwarded for" chain, not discard the entire chain.